### PR TITLE
Added Details on Physics engines

### PR DESCRIPTION
**Pull request by BabylonJSGuide**
Made a start on expanding on physics' engines information. Over next few weeks will add more explanation.

### CODE REVIEW

**Comment by RaananW** [comment]
This is really great! 
My comments are minor (only suggestions) and should not affect merging this

**Comment by deltakosh** [comment]
Pinging @TrevorDev  for a quick review :)

**Review by TrevorDev — COMMENTED** [review]
@BabylonJSGuide  Awesome, this looks great! I see you noted in the differences between the engines that ammo behaves differently in some cases that oimo/cannon, these seem like they should be bugs in the ammoJS plugin, don't you agree?

It looks like the main issue is the contact point for impulses which is why you create separate playground for oimo/cannon vs ammo. I should fix this. Are there other major differences I should fix?

Recommend: remove separate playgrounds and "** differences **" related to this, even the oimo force requiring a very high number vs ammo/cannon seems like a bug @RaananW, maybe you can chime in if you agree for that difference. The only valid differences should be features that are/are not supported by each engine along with minor differences physics calculation output.

**Comment by BabylonJSGuide** [comment]
@TrevorDev @RaananW 
You will also see significant differences in the joints page and the compounds page.
 
**Joints**

> Replacing the sphere mesh imposter with a cylinder imposter produces further changes. 
> For `Cannon.js` you need to swap the y and z values in the mainAxis vector only, for a correct rotation direction

The swapping of y and z could probably be done in the Cannon plugin. I note it is done on some occasions in the plugin but not others.

**Compounds**
If you take either of the Ammo or Oimo playgrounds and just replace with the CannonJSPlugin I could not get the skull compound to work with Cannon. The only example I found of a compound for Cannon is https://playground.babylonjs.com/#PRHF00#6 and I just used this as a basis for the Cannon playground.

Do you agree that a merge of this PR should be held until plugins are aligned? Would these go in as a breaking change? Your thoughts.

**Comment by RaananW** [comment]
The oimo impulse application was set to stay backwards compatible. If it is agreed to fix it, i think you should do that. 
I think the z and y for cannon should be swapped internally in the engine plugin and not by the user.
I also agree with @TrevorDev regarding separated playgrounds. whenever possible we should have the same playgrounds for all engines

**Comment by TrevorDev** [comment]
@BabylonJSGuide I think this PR should be held until these bugs are fixed as we don't wan't people coding against api's that will change soon. If we think we know what bugs we need to fix already, this doc could be updated assuming those will be fixed and checked in. 

@RaananW yea, I would prefer to fix the oimo strong force. 

I'm hoping by the end of next week I will be able to take care of all the physics related issues that popped up recently.

**Comment by BabylonJSGuide** [comment]
Also on Oimo setMotor this seems to require both parameters with the second one having the biggest affect. Other two work OK with one parameter.

Will change the PR on the assumption that the plugins will all react in the same way. 

@TrevorDev once changes are made to the plugins I am happy to test them out.

**Comment by BabylonJSGuide** [comment]
Issues now referenced in [Babylon.js #5736](https://github.com/BabylonJS/Babylon.js/issues/5736)

**Review by deltakosh — APPROVED** [review]

**Comment by deltakosh** [comment]
@TrevorDev : good to go?

File: content/How_To/Physics/Joints.md [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+The method to form a joint and connect one body (main) to a second body (connected) is 
+
+```javascript
+var joint = new BABYLON.PhysicsJoint(BABYLON.PhysicsJoint.TYPE_OF_JOINT, jointData);
+
+mainImpostor.addJoint(connectedImpostor, joint);
+```
+where the `jointData` object contains the properties for the joint.
+
+**Note** `Ammo.js` for some as yet unexplained reason sometimes does not apply an impulse if triggered through a button. To ensure triggering an added direct impulse with a smallish size (that does not overcome inertia) is applied after joint construction. It is hoped that this issue can be resolved in the future.
```
Comment (TrevorDev) [inline content/How_To/Physics/Joints.md]: Is this not fixed now?

**Comment by BabylonJSGuide** [comment]
A couple of points to check with @TrevorDev and @deltakosh before final version commited.

1. Since Oimo does not have a native applyForce BJS applyForce is applyImpulse in Oimo so same value has a much bigger effect in Oimo than Cannon and Ammo. 
https://www.babylonjs-playground.com/#RHBQY9#1
Should this remain for backward capability?

2. Cannon places box other way up https://www.babylonjs-playground.com/#F15U0G#62 is this a big enough issue to need changes

File: content/How_To/Physics/Joints.md (hunk 1 above)
Comment (TrevorDev) [reply content/How_To/Physics/Joints.md]: Other than this, everything looks good to me. Thanks @BabylonJSGuide

**Comment by BabylonJSGuide** [comment]
All other PG now work consistently with all three plugins.

File: content/How_To/Physics/Joints.md (hunk 1 above)
Comment (BabylonJSGuide) [reply content/How_To/Physics/Joints.md]: Sorry yes above issue fixed

**Comment by BabylonJSGuide** [comment]
Also want to give docs a final check tomorrow to make sure I haven't made any mistakes as it is getting late here. Should be able to finalise tomorrow.

**Comment by TrevorDev** [comment]
1. Yes, I think it should remain for back compat reasons since oimo doesn't support it.
2. I think that issue is not to do with the box orientation, it seems cannon's axis's don't work the same way as oimo/ammo and I'm unable to find documentation that clearly describes how it works so I can translate it to work similarly. From my testing the root issue seems to come from  the axis not respecting inverse direction. See https://www.babylonjs-playground.com/#F15U0G#62, if you change main axis from 0,1,0 to 0,-1,0 nothing changes but when switching to ammo the direction changes. So Im not sure that it is possible to translate so this might have to be a won't fix.

**Comment by BabylonJSGuide** [comment]
Final corrections made if @TrevorDev and @deltakosh happy are happy then PR can be merged.
Thank you Trevor for all the bug corrections and making the plugins consistent.

**Comment by deltakosh** [comment]
Good for me!

**Review by TrevorDev — APPROVED** [review]
👍

**Comment by BabylonJSGuide** [comment]
merged

**Comment by deltakosh** [comment]
Thanks a ton @BabylonJSGuide

**Pull request merged**